crates/solidity/inputs/language/src/definition.rs

```diff
@@ -0,0 +1,3854 @@
+                                )
+                            ]
+                        )
+                    ]
+                ),
+                Topic(
+                    title = "Modifiers",
+                    items = [
+                        Struct(
+                            name = ModifierDefinition,
+                            fields = (
+                                modifier_keyword = Required(Terminal([ModifierKeyword])),
+                                name = Required(Terminal([Identifier])),
+                                parameters =
+                                    Optional(reference = NonTerminal(ParametersDeclaration)),
+                                attributes = Required(NonTerminal(ModifierAttributes)),
+                                body = Required(NonTerminal(FunctionBody))
+                            )
+                        ),
+                        Repeated(
+                            name = ModifierAttributes,
+                            repeated = NonTerminal(ModifierAttribute),
+                            allow_empty = true
+                        ),
+                        Enum(
+                            name = ModifierAttribute,
+                            variants = [
+                                Variant(
+                                    name = Override,
+                                    fields = (specifier = Required(NonTerminal(OverrideSpecifier)))
+                                ),
+                                Variant(
+                                    name = Virtual,
+                                    enabled_in = "0.6.0",
+                                    fields = (keyword = Required(Terminal([VirtualKeyword])))
+                                )
+                            ]
+                        ),
+                        Struct(
+                            name = ModifierInvocation,
+                            fields = (
+                                name = Required(NonTerminal(IdentifierPath)),
+                                arguments = Optional(reference = NonTerminal(ArgumentsDeclaration))
+                            )
+                        )
+                    ]
+                ),
+                Topic(
+                    title = "Events",
+                    items = [
+                        Struct(
+                            name = EventDefinition,
+                            fields = (
+                                event_keyword = Required(Terminal([EventKeyword])),
+                                name = Required(Terminal([Identifier])),
+                                parameters =
+                                    Optional(reference = NonTerminal(EventParametersDeclaration)),
+                                anonymous_keyword =
+                                    Optional(reference = Terminal([AnonymousKeyword])),
+                                semicolon = Required(Terminal([Semicolon]))
```

Comment:
I prefer this to be a `TerminatedBy` construction so that the error recovery can pick it up.

Reply:
Discussing offline.

Reply:
Added `error_recovery` field.